stores/migrations.go

```diff
@@ -206,7 +204,14 @@ func performMigrations(db *gorm.DB, logger glogger.Interface) error {
 // initSchema is executed only on a clean database. Otherwise the individual
 // migrations are executed.
 func initSchema(tx *gorm.DB) error {
-	err := tx.AutoMigrate(tables...)
+	// Setup join tables.
+	err := setupJoinTables(tx)
```

Comment:
Trick was to execute this _before_ doing auto migrations, I'm not sure why to be honest... I wanted to draw attention to this because this is something I'm letting slide for the most part because it just works, but I don't understand why setting up the join tables fail if you do it after doing auto migrations...

Reply:
I'm pretty sure it fails because `AutoMigration` reads the `many2many` annotation and creates the default join table since it's not aware of your custom type. But if you call `SetupJoinTable` on your type the schema is created which is then used by `AutoMigrate` instead.

But that's just a wild guess since there isn't a lot of documentation about it.